addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1476,18 +1478,27 @@ const Wysiwyg = Widget.extend({
         if ($colorpickerGroup.length) {
             this._createPalette();
         }
-        // we need the Timeout to be sure the editable content is loaded
-        // before calculating the scrollParent() element.
-        setTimeout(() => {
-            const scrollableContainer = this.$el.scrollParent();
-            if (!options.snippets && scrollableContainer.length) {
-                this.odooEditor.addDomListener(
-                    scrollableContainer[0],
-                    'scroll',
-                    this.odooEditor.updateToolbarPosition.bind(this.odooEditor),
-                );
-            }
-        }, 0);
+        if (!options.snippets) {
+            const isEditableInSubTree = memoize(this._isEditableInSubTree.bind(this));
+            this._onScroll = ev => {
+                if (isEditableInSubTree(ev.target)) {
+                    this.odooEditor.updateToolbarPosition();
+                }
+            };
+            // Scroll event does not bubble.
+            document.addEventListener('scroll', this._onScroll, true);
+        }
+    },
+    /**
+     * Whether the editable is part of the subtree rooted at 'root'.
+     *
+     * @private
+     * @param {Node} root
+     * @returns {Boolean}
+     */
+    _isEditableInSubTree(root) {
+        const isInTree = node => node && (node === root || isInTree(node.parentNode, root));
+        return isInTree(this.odooEditor.editable);
```

Comment:
Also known as `return root.contains(this.odooEditor.editable)`.
My bad. I will change it after RD TESTING BE is green...

Reply:
Well, now is the time, then... :p